File: api/models/interviewTemplate.js
let templates = [];
let idCounter = 1;

const MOCK_TEMPLATES_DATA = [
  {
    jobTitle: 'Senior Frontend Developer',
    category: 'Software Engineering',
    experienceLevel: 'Senior',
    questions: [
      'Explain the difference between `let`, `const`, and `var` in JavaScript.',
      'What are Angular Signals and how do they improve change detection?',
      'Describe your experience with state management libraries like NgRx or Redux.',
      'How would you optimize a slow-loading web page?',
      'Discuss the importance of web accessibility (a11y).'
    ],
    type: 'technical',
    company: 'HussnainTechVertex Pvt Ltd.',
    authorId: '1', // Corresponds to super-admin user
    authorName: 'Hussnain',
  },
  {
    jobTitle: 'UX/UI Designer',
    category: 'UX/UI Design',
    experienceLevel: 'Mid-Level',
    questions: [
      'Walk me through your design process from concept to final handoff.',
      'How do you handle feedback from non-designers?',
      'What is your experience with user research and usability testing?',
      'Can you show me a project you are particularly proud of and explain why?'
    ],
    type: 'behavioral',
    company: 'HussnainTechVertex Pvt Ltd.',
    authorId: '1',
    authorName: 'Hussnain',
  },
  {
    jobTitle: 'Product Manager',
    category: 'Product Management',
    experienceLevel: 'Lead',
    questions: [
        'How do you prioritize features for a product roadmap?',
        'Describe a time you had to make a difficult decision with limited data.',
        'How do you define and measure success for a product?',
    ],
    type: 'situational',
    company: 'HussnainTechVertex Pvt Ltd.',
    authorId: '1',
    authorName: 'Hussnain',
  }
];

const initializeDefaultTemplates = () => {
    templates = MOCK_TEMPLATES_DATA.map(template => ({
        ...template,
        id: String(idCounter++),
        createdAt: new Date().toISOString()
    }));
};

initializeDefaultTemplates();

const InterviewTemplate = {
    findAll: async () => {
        return templates;
    },
    findById: async (id) => {
        return templates.find(t => t.id === id);
    },
    create: async (templateData) => {
        const newTemplate = {
            ...templateData,
            id: String(idCounter++),
            createdAt: new Date().toISOString()
        };
        templates.push(newTemplate);
        return newTemplate;
    },
    delete: async (id) => {
        const index = templates.findIndex(t => t.id === id);
        if (index > -1) {
            templates.splice(index, 1);
            return true;
        }
        return false;
    }
};

module.exports = InterviewTemplate;

File: backend_python/app/models/interview_template.py
from datetime import datetime
from sqlalchemy import String, JSON, DateTime
from sqlalchemy.orm import Mapped, mapped_column

from app.db.session import Base


class InterviewTemplate(Base):
    __tablename__ = "interview_templates"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    job_title: Mapped[str] = mapped_column(String(255), nullable=False)
    category: Mapped[str] = mapped_column(String(128), nullable=False)
    experience_level: Mapped[str] = mapped_column(String(128), nullable=False)
    questions: Mapped[list[str]] = mapped_column(JSON, default=list)
    type: Mapped[str] = mapped_column(String(64), nullable=False)
    company: Mapped[str | None] = mapped_column(String(255))
    author_id: Mapped[int] = mapped_column(index=True)
    author_name: Mapped[str] = mapped_column(String(255))
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
